lib/codecs/HttpUrlEncoder.js

```diff
@@ -5,29 +5,37 @@ import {Token, Type} from 'cborg';
 import {CborldEncoder} from './CborldEncoder.js';
 
 export class HttpUrlEncoder extends CborldEncoder {
-  constructor({value, secure} = {}) {
+  constructor({value, secure, compressionMap} = {}) {
     super();
     this.value = value;
     this.secure = secure;
+    this.compressionMap = compressionMap;
   }
 
   encode() {
     const {value, secure} = this;
     const length = secure ? 'https://'.length : 'http://'.length;
     const entries = [
-      new Token(Type.uint, secure ? 2 : 1),
+      new Token(Type.uint, secure ? this.compressionMap.get('https://') : this.compressionMap.get('http://')),
```

Comment:
This looks like it will cause a lint error (over 80 chars).

```suggestion
      new Token(Type.uint, secure ?
        this.compressionMap.get('https://') : this.compressionMap.get('http://')),
```